src/Assets/Manifest.php

```diff
@@ -0,0 +1,138 @@
+<?php
+namespace WPSentry\Assets;
+
+// Exit if plugin isn't running
+defined( 'WP_SENTRY_EXISTS' ) || exit;
+
+/**
+ * Class Manifest.
+ *
+ * A class for fetching dynamic assets from a json manifest.
+ *
+ * @package WPSentry\Assets;
+ * @since   3.0.0
+ */
+
+class Manifest{
+
+  /**
+   * The content of the manifest we are reading from.
+   *
+   * @since 3.0.0
+   * @var string
+   */
+  private $manifest_content;
+
+  /**
+   * The path of the manifest file we are reading from.
+   *
+   * @since 3.0.0
+   * @var string
+   */
+  private $manifest_path;
+
+  /**
+   * The URL of the plugin's `dist` folder where our final assets live.
+   *
+   * @since 3.0.0
+   * @var string
+   */
+  private $dist_url;
+
+
+  /**
+   * Class Constructor
+   */
+  public function __construct(){
+
+    $this->manifest_content = null;
+    $this->manifest_path = defined( 'WP_SENTRY_ASSET_MANIFEST_PATH') ? WP_SENTRY_ASSET_MANIFEST_PATH : null;
```

Comment:
@killua99 
I like the `??` operator but have found it to be inconsistent because it only resolves the right half if the left is `null`. It does not resolve if the left half is empty, or undefined, or false. Therefore I have stopped using as much in my codebases. Thoughts?

Reply:
This is true, but for this constants situation it apply perfectly, we just want to guarantee that "isset" that constant and has a value. if is not defined will be null, if is defined it will have a value, false, empty or what ever. You're not checking the type of definition in here. That's why I thought about this null coalescing operator